bftengine/src/bftengine/messages/PrePrepareMsg.cpp

```diff
@@ -77,11 +73,21 @@ void PrePrepareMsg::validate(const ReplicasInfo& repInfo) const {
 }
 
 PrePrepareMsg::PrePrepareMsg(ReplicaId sender, ViewNum v, SeqNum s, CommitPath firstPath, bool isNull, size_t size)
-    : MessageBase(
-          sender,
-          MsgCode::PrePrepare,
-          (((size + sizeof(PrePrepareMsgHeader)) < maxSizeOfPrePrepareMsg()) ? (size + sizeof(PrePrepareMsgHeader))
-                                                                             : maxSizeOfPrePrepareMsg()))
+    : PrePrepareMsg(sender, v, s, firstPath, "", isNull, size) {}
+
+PrePrepareMsg::PrePrepareMsg(ReplicaId sender,
+                             ViewNum v,
+                             SeqNum s,
+                             CommitPath firstPath,
+                             const std::string& spanContext,
+                             bool isNull,
+                             size_t size)
+    : MessageBase(sender,
+                  MsgCode::PrePrepare,
+                  spanContext.size(),
+                  (((size + sizeof(Header)) < maxMessageSize<PrePrepareMsg>())
+                       ? (size + sizeof(Header))
+                       : maxMessageSize<PrePrepareMsg>() - spanContext.size()))
```

Comment:
Just by looking at hte size calculation logic, I can't be 100% it is equivalent to the old logic:
```

          (((size + sizeof(PrePrepareMsgHeader)) < maxSizeOfPrePrepareMsg()) ? (size + sizeof(PrePrepareMsgHeader))
                                                                             : maxSizeOfPrePrepareMsg()))
```

Reply:
I think it is the same with one exception which the span size.
However, I might miss something. Probably it worth it to have another pair of eyes on it.
@andrewjstone , could you have a look?